rio_tiler/reader.py

```diff
@@ -23,6 +23,16 @@
 from rio_tiler.utils import get_vrt_transform, has_alpha_band, non_alpha_indexes
 
 
+class Options(TypedDict):
+    """Reader Options."""
+
+    nodata: Optional[NoData]
```

Comment:
I think we want `NotRequired` here isntead https://stackoverflow.com/questions/72326834/how-to-have-optional-keys-in-typeddict

Reply:
Also note:

> Note: If you import Required or NotRequired from typing_extensions (on Python <3.11), you should also import TypedDict from typing_extensions.